pkg/frontend/querymiddleware/querysharding.go

```diff
@@ -496,7 +508,7 @@ type annotationAccumulator struct {
 	infos    *sync.Map
 }
 
-func newAnnotationAccumulator() *annotationAccumulator {
+func NewAnnotationAccumulator() *annotationAccumulator { //nolint:revive
```

Comment:
Should we just export `annotationAccumulator`? Seems weird to export `NewAnnotationAccumulator` but not the type it returns.